atlas_checks.xml

```diff
@@ -0,0 +1,213 @@
+        -->
+        <module name="LineLength">
+            <property name="max" value="120"/>
+        </module>
+        <module name="MethodLength"/>
+        <!--<module name="ParameterNumber"/>-->
+        <module name="ParameterNumber">
+            <property name="max" value="23"/>
+        </module>
+
+        <module name="ParameterNumber">
+            <property name="max" value="23"/>
+        </module>
+
+
+        <!-- Checks for whitespace                               -->
+        <!-- See http://checkstyle.sf.net/config_whitespace.html -->
+        <module name="EmptyForIteratorPad"/>
+        <module name="GenericWhitespace"/>
+        <module name="MethodParamPad"/>
+        <module name="NoWhitespaceAfter"/>
+        <module name="NoWhitespaceBefore"/>
+        <module name="OperatorWrap">
+            <property name="tokens" value="QUESTION"/>
+            <property name="option" value="eol"/>
+        </module>
+        <!--<module name="OperatorWrap"/>-->
+        <module name="ParenPad"/>
+        <module name="TypecastParenPad"/>
+        <module name="WhitespaceAfter"/>
+        <!--<module name="WhitespaceAround"/>-->
+        <module name="WhitespaceAround">
+            <property name="allowEmptyConstructors" value="true"/>
+            <property name="allowEmptyMethods" value="true"/>
+            <property name="allowEmptyLambdas" value="true"/>
+        </module>
+
+        <!-- Modifier Checks                                    -->
+        <!-- See http://checkstyle.sf.net/config_modifiers.html -->
+        <module name="ModifierOrder"/>
+        <module name="RedundantModifier"/>
+
+        <!-- Checks for blocks. You know, those {}'s         -->
+        <!-- See http://checkstyle.sf.net/config_blocks.html -->
+        <module name="AvoidNestedBlocks"/>
+        <module name="EmptyBlock">
+            <property name="option" value="text"/>
+        </module>
+
+        <module name="LeftCurly"/>
+        <module name="NeedBraces"/>
+        <module name="RightCurly"/>
+
+        <!-- Checks for common coding problems               -->
+        <!-- See http://checkstyle.sf.net/config_coding.html -->
+        <!--
+        <module name="AvoidInlineConditionals"/>
+        -->
+        <module name="EmptyStatement"/>
+        <module name="EqualsHashCode"/>
+        <module name="IllegalInstantiation"/>
+        <!--<module name="HiddenField"/>-->
+        <module name="HiddenField">
+            <property name="ignoreConstructorParameter" value="true"/>
+            <property name="ignoreSetter" value="true"/>
+            <property name="setterCanReturnItsClass" value="true"/>
+            <property name="tokens" value="VARIABLE_DEF"/>
+        </module>
+        <module name="InnerAssignment"/>
+        <!--<module name="MagicNumber"/>-->
+        <module name="MagicNumber">
+            <property name="ignoreNumbers" value="-1, 0, 0.1, 0.25, 0.75, 1, 2, 3, 4, 5, 6, 10, 60, 100, 1000, 45000"/>
```

Comment:
Why is it ignoring these numbers specifically?

Reply:
Checkstyle by default ignores -1, 0, 1 and 2. All other magic numbers are flagged and those are the ones we use in our code. I added them here because in the context where they appear it’s quite clear what they represent (e.g. `setExpressionCutoff(0.1)` or something along those lines).